utide/harmonics.py

```diff
@@ -26,7 +26,8 @@
 def linearized_freqs(tref):
     astro, ader = ut_astron(tref)
     freq = const.freq.copy()
-    freq[not_shallow] = np.dot(const.doodson[not_shallow, :], ader) / 24
+    selected = np.dot(const.doodson[not_shallow, ...], ader) / 24
```

Comment:
ignore my ignorance here, but what does `...` do?

Reply:
Good catch--I hadn't noticed that.

It means full slices of all remaining dimensions.  It's not needed here. The one advantage of keeping the `:` in the index expression is that it helps readability by showing that `const.doodson` is 2-D.  So I would change the `...` back to `:`.

Reply:
Indeed it does not do anything beyond obfuscate the `const.doodson` shape.
I left it there while debugging, we don't need it.